ts/nni_manager/common/log.ts

```diff
@@ -1,141 +1,119 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import fs from 'fs';
-import { Writable } from 'stream';
-import util from 'util';
-
-/* log level constants */
+/**
+ *  Python-like logging interface.
+ *
+ *      const logger = getLogger('moduleName');
+ *      logger.info('hello', { to: 'world' });
+ *
+ *  Outputs:
+ *
+ *      [1970-01-01 00:00:00] INFO (moduleName) hello { to: 'world' }
+ **/
 
-export const DEBUG = 10;
-export const INFO = 20;
-export const WARNING = 30;
-export const ERROR = 40;
-export const CRITICAL = 50;
+import util from 'util';
 
-export const TRACE = 1;
-export const FATAL = 50;
+import globals from 'common/globals';
 
-const levelNames = new Map<number, string>([
-    [CRITICAL, 'CRITICAL'],
-    [ERROR, 'ERROR'],
-    [WARNING, 'WARNING'],
-    [INFO, 'INFO'],
-    [DEBUG, 'DEBUG'],
-    [TRACE, 'TRACE'],
-]);
+const levelNameToValue = { trace: 0, debug: 10, info: 20, warning: 30, error: 40, critical: 50 } as const;
 
-/* global_ states */
+const loggers: Record<string, Logger> = {};
 
-let logLevel: number = 0;
-const loggers = new Map<string, Logger>();
+export function getLogger(name: string): Logger {
+    if (loggers[name] === undefined) {
+        loggers[name] = new Logger(name);
+    }
+    return loggers[name];
+}
 
-/* major api */
+/**
+ *  A special logger prints to stderr when the logging system has problems.
+ *  For modules that are responsible for handling logger errors.
+ **/
+export function getRobustLogger(name: string): Logger {
```

Comment:
I didn't find where this is used.

Reply:
In shutdown manager.
This is a split PR.